ftw/flamegraph/__init__.py

```diff
@@ -72,20 +81,45 @@ def publish_module_standard(
 Publish.publish_module_standard = publish_module_standard
 
 
+def make_svg(sampler):
+    tempdir = tempfile.mkdtemp()
+    flame_file_path = os.path.join(tempdir, 'profile.flame')
+    svg_file_path = os.path.join(tempdir, 'profile.svg')
+
+    with open(flame_file_path, "wb") as f:
+        f.write(sampler.folded_stacks().encode('utf-8'))
+
+    os.system(
+        '%(flamegraph_script)s --egghash --color rainbow %(flame_file)s > '
+        '%(svg_file)s' % (dict(
+            flamegraph_script=flamegraph_script,
+            flame_file=flame_file_path,
+            svg_file=svg_file_path)))
+
+    return svg_file_path
+
+
 @contextmanager
-def profile(interval):
+def profile(interval, print_summary=False):
+    global sampler
+    if sampler is None:
+        sampler = Sampler()
+
     sampler.reset(interval=interval)
     sampler.start()
     start = time.time()
-    yield
+    yield sampler
     end = time.time()
     sampler.stop()
 
-    logger.info("Samples taken: %s, sample time: %s ms, request time: %s ms" % (
+    message = "Samples taken: %s, sample time: %s ms, request time: %s ms" % (
```

Comment:
I would rename "request time" to "duration" because profiling is no longer limited to Zope requests.